### PR TITLE
  ✨ feat: optimize log reporting performance with batch sending

### PR DESCRIPTION
**Pull request by li-zhixin**
## Summary
  Introduces a batching mechanism for log reporting to reduce network requests and improve performance.

  ## Changes
  - Add LogBatcher class to implement log batching logic
  - Support batching strategy based on entry count and payload size limits
  - Add configuration options: `batchLogs`, `batchLogsSize`, `batchPayloadLimit`
  - Implement `flushLogs()` method to manually flush pending logs
  - Update TypeScript type definitions and documentation
  - Add comprehensive unit tests covering various batching scenarios

  ## Benefits
  - Reduces network overhead by consolidating multiple log entries into single requests
  - Configurable batching parameters to balance between latency and throughput
  - Maintains backward compatibility

## Summary by CodeRabbit

* **New Features**
  * Optional log batching with configurable enable, batch size, and payload limit
  * Added flushLogs() API to manually flush pending batched logs; batches also flush automatically on size/payload limits or when finishing a launch

* **Documentation**
  * Documented batch logging options, defaults, and usage examples

* **Tests**
  * Added comprehensive tests for batching behavior, limits, flush semantics, and edge cases

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by li-zhixin** [comment]
@HardNorth  I'm sorry to bother you again. I've reimplemented it based on your feedback. Could you please take another look?

File: pnpm-lock.yaml
Comment (HardNorth) [inline pnpm-lock.yaml:1]: Please remove, we don't use `pnpm`.

**Review by HardNorth — CHANGES_REQUESTED** [review]
Please fix tests.

**Comment by li-zhixin** [comment]
> Please fix tests.

done.

**Comment by li-zhixin** [comment]
I'm sorry to interrupt again.  Can someone kind help me look at it again

File: lib/logs/batcher.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+  /**
+   * Internal method to append a log request with known size.
+   * @private
+   * @param {number} size - calculated size of the log request in bytes
+   * @param {Object} logReq - log request object with payload and optional file
+   * @returns {Array|null} batch array if ready to send, null otherwise
+   */
+  appendInternal(size, logReq) {
+    if (size >= this.payloadLimit) {
```
Comment (maria-hambardzumian) [inline lib/logs/batcher.js:29]: @li-zhixin Could you please also consider adding a warning message when a single log entry exceeds payloadLimit? This would help developers identify oversized logs during debugging.

File: lib/commons/config.js [hunk 2]
```diff
@@ -116,6 +117,9 @@ const getClientConfig = (options) => {
       launchUuidPrint: options.launchUuidPrint,
       launchUuidPrintOutput,
       skippedIsNotIssue: !!options.skippedIsNotIssue,
+      batchLogs: options.batchLogs ?? false,
+      batchLogsSize: options.batchLogsSize ?? MAX_LOG_BATCH_SIZE,
```
Comment (maria-hambardzumian) [inline lib/commons/config.js]: Could you please consider adding validation for batchLogsSize and batchPayloadLimit config values? Currently, invalid values (≤ 0, non-numbers, null) could cause infinite buffering or unpredictable behavior. A simple check in the constructor would prevent silent issues.

File: __tests__/log-batcher.spec.js [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+        },
+      };
+
+      const result = batcher.append(logReq);
+
+      expect(result).toBeNull();
+      expect(batcher.batchSize).toBe(1);
+    });
+  });
+});
```
Comment (maria-hambardzumian) [inline __tests__/log-batcher.spec.js:350]: It would be better if we could add tests for failure paths and invalid inputs for better consistency. Currently, tests only cover happy paths.

**Review by maria-hambardzumian — CHANGES_REQUESTED** [review]

**Comment by li-zhixin** [comment]
@maria-hambardzumian  Thank you for your time in pointing out the problem,  I've fixed it.

**Comment by li-zhixin** [comment]
@maria-hambardzumian I'm sorry to bother you, could you please take another look?

**Review by maria-hambardzumian — APPROVED** [review]

**Comment by maria-hambardzumian** [comment]
@AmsterGet can you take a look?

**Comment by li-zhixin** [comment]
Sorry to bother you again. If there are any issues, please let me know and I'll do my best to fix them.

**Comment by li-zhixin** [comment]
I'm sorry for bothering you again and again, but could anyone please help move this along? Even a rejection would be better than no response at all.  @maria-hambardzumian  @HardNorth  @stevez  @lexecon

**Comment by li-zhixin** [comment]
Just leaving a gentle follow-up here. Whenever it fits your schedule, I’d appreciate any feedback on this PR. If any changes are needed, I’m happy to make them. Thank you very much for
  your time.

**Comment by AmsterGet** [comment]
Hello @li-zhixin !
Sorry for the silence :(
The review of this PR has stalled on my end.
I was ill for quite some time in February and March, which is why the process has been delayed. I hope you’ll understand.
I’ll have time next week to look through your contribution, and I’ll be sure to get back to you with my feedback.
Thank you for your efforts, and I apologize once again for the delay.
